api/src/main/java/com/example/api/service/map/ChapterService.java

```diff
@@ -141,4 +167,25 @@ public void editChapter(EditChapterForm editChapterForm) throws RequestValidatio
 
         chapter.setActivityMap(chapterMap);
     }
+
+    public RequirementResponse getRequirementsForChapter(Long chapterId) throws EntityNotFoundException {
+        Chapter chapter = chapterRepo.findChapterById(chapterId);
+        chapterValidator.validateChapterIsNotNull(chapter, chapterId);
+
+        List<? extends RequirementDTO<?>> requirements = chapter.getRequirements()
+                .stream()
+                .map(Requirement::getResponse)
+                .toList();
+        return new RequirementResponse(chapter.getIsBlocked(), requirements);
+    }
+
+    public void addRequirementToChapter(ChapterRequirementForm form) throws RequestValidationException {
+        Chapter chapter = chapterRepo.findChapterById(form.getChapterId());
+        Boolean isBlocked = form.getIsBlocked();
+        if(isBlocked != null) {
+            chapter.setIsBlocked(isBlocked);
+        }
+        List<RequirementForm> requirementForms = form.getRequirements();
+        requirementService.updateRequirements(requirementForms);
+    }
```

Comment:
Wydaje mi się, że ta funkcja służy jedynie do update'u wymagań, więc mogłaby się nazywać updateRequirementToChapter(), czy się mylę? Wiedzę, że w TaskService jest funkcja addRequirementToActivity, która działa podobnie. Bo z tego co rozumiem to wymagania są tworzone na starcie (wszystkie rodzaje) i przy ich dodawaniu tak naprawdę je updatujemy, tak?

Reply:
Tak, masz rację nazewnictwo troche tu leży, tak samo jest w nazwie endpointów